piptools/scripts/utils.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from ..logging import log
+from ..utils import PIP_VERSION
+
+
+def drop_unsupported_pip_args(args: list[str]) -> None:
```

Comment:
I just realized that this mutates incoming object. It's a dangerous practice that causes difficult-to-follow mutation sequences. It's not really used in Python. Let's rewrite it to return something that can then be assigned on the calling side. Otherwise, the calling side wouldn't be expecting the var they pass in to suddenly become different in terms of content.